### PR TITLE
ARROW-16143: [Java] Upgrade jackson dependencies CVE-2020-36518

### PR DESCRIPTION
**Pull request by davisusanibar**
[CVE-2020-36518](https://github.com/advisories/GHSA-57j2-w4cx-62h2): Deeply nested json in jackson-databind.

Solved based on [Jackson release 2.13](https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.13):
````
jackson-databind 2.13.2.2 (28-Mar-2022) -- with jackson-bom version 2.13.2.20220328
````

Before the change: ` mvn compile dependency:tree -Drat.skip=true | grep databind` :
````
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:test
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:compile
````

After the change:
````
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.4:test
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.2.2:compile
````

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <dep.guava.version>30.1.1-jre</dep.guava.version>
     <dep.netty.version>4.1.72.Final</dep.netty.version>
     <dep.jackson.version>2.11.4</dep.jackson.version>
+    <dep.jackson-bom.version>2.13.2.20220328</dep.jackson-bom.version>
```
Comment (lidavidm) [inline java/pom.xml:38]: Why the discrepancy, why can't we just edit dep.jackson.version above?

Comment (davisusanibar) [reply java/pom.xml:38]: I just updated `<dep.jackson.version>2.13.2</dep.jackson.version>` but consider if we maintain that with only this parameter version all libraries are going be updated to 2.13.2, but, for jackson-databind we need 2.13.2.2 and this is handled by [jackson BOM](https://github.com/FasterXML/jackson-bom/blob/jackson-bom-2.13.2.20220328/pom.xml#L62) `<dep.jackson-bom.version>2.13.2.20220328</dep.jackson-bom.version>`

Comment (lidavidm) [reply java/pom.xml:38]: I don't think we want to version the Jackson libraries separately though. If we're upgrading one we should upgrade all.

Comment (davisusanibar) [reply java/pom.xml:38]: At summary we need two variables:
- For jackson-databind: cve 2.13.2.2 (We have two option: modify all pom.xml and set this version OR delegate this to jackson-bom)
- For jackson-core, jackson-annotation, jackson-dataformat-yaml, jackson-datatype-jsr310: 2.13.2

Comment (lidavidm) [reply java/pom.xml:38]: Ah, they only published revisions for one library…Alright then. Can we make it `dep.jackson.databind.version` or something and note that upstream versioning is out of sync? It's unclear what `bom` is supposed to control.

Comment (lidavidm) [reply java/pom.xml:38]: Oh, sorry, we depend on `bom` indeed. Alright then.

**Comment by lidavidm** [comment]
@github-actions crossbow submit java-jars verify-rc-source-java-*

**Pull request closed without merging**